assignment-1/src/task4.py

```diff
@@ -1,97 +0,0 @@
-import time
-import contextlib
-import inspect
-import atexit
-import traceback
-import datetime
-import io
-
-def exit_hd():
-    """To print the ranks of each program just before the program is finished"""
-    sorted_ranks = sorted(ranks.items(), key=lambda x: x[1])
-    if len(sorted_ranks) > 0:
```

Comment:
You can omit comparison with 0

Reply:
This comparison is necessary in one case, Lets say we have decorated all our function in the main.py with the decorator_2 but not the other decorator_3, decorator_4 then the program will print "PROGRAM	|	RANK	|	TIME ELAPSED" just before it is finished executing.

You can replicate it by just replacing all the decorators above the function with the decorator_2, without you comparison of 0, you will find that there is the printing of "PROGRAM	|	RANK	|	TIME ELAPSED", which is not necessary.

Reply:
What I meant: "if len(sorted_ranks) > 0:" is equivalent to "if Len(sorted_ranks):".

Reply:
Understood and Noted )